packages/sonarwhal/src/lib/connectors/local/local.ts

```diff
@@ -247,6 +219,50 @@ export default class LocalConnector implements IConnector {
      * ------------------------------------------------------------------------------
      */
 
+    public async fetchContent(target: string) {
+        /*
+         * target can have one of these forms:
+         *   - /path/to/file
+         *   - C:/path/to/file
+         *   - file:///path/to/file
+         *   - file:///C:/path/to/file
+         *
+         * That's why we need to parse it to an URL
+         * and then get the path string.
+         */
+        const uri: url.URL = getAsUri(target);
+        const filePath: string = getAsPathString(uri);
+        const rawContent = await fs.readFile(filePath);
+        const contentType = getContentTypeData(null, filePath, null, rawContent);
+        let content = '';
+
+        if (isTextMediaType(contentType.mediaType)) {
+            content = rawContent.toString(contentType.charset);
+        }
+
+        // Need to do some magic to create a fetch::end::*
+        return {
+            element: null,
+            request: null,
+            resource: url.format(getAsUri(filePath)),
+            response: {
+                body: {
+                    content,
+                    rawContent,
+                    rawResponse() {
+                        return Promise.resolve(rawContent);
+                    }
+                },
+                charset: contentType.charset,
+                headers: null,
+                hops: [],
+                mediaType: contentType.mediaType,
+                statusCode: 200,
+                url: filePath
+            }
+        };
+    }
+
```

Comment:
This commit should start with `New:` as we are adding a new feature.

Reply:
done!